<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from summit_xls_std.urdf.xacro      | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="summit_xl">
  <!-- ***************** -->
  <!-- Imported elements -->
  <!-- ***************** -->
  <!-- First we import all posible elements defined in the urdf.xacro files. All these elements are defined as macro:xacros -->


  <!-- BASE_LINK -->
  <link name="base_x_slide_link" />
  <joint name="base_x_slide_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
      <parent link="base_x_slide_link"/>
      <child link="base_y_slide_link"/>
      <axis xyz= "1 0 0"/>

  </joint>
  <link name="base_y_slide_link" />
  <joint name="base_y_slide_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
      <parent link="base_y_slide_link"/>
      <child link="base_z_hinge_link"/>
      <axis xyz= "0 1 0"/>

  </joint>
  <link name="base_z_hinge_link" />
  <joint name="base_z_hinge_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0 0 0"/>
      <parent link="base_z_hinge_link"/>
      <child link="base"/>
      <axis rpy="0 0 0" xyz="0 0 1"/>

  </joint>
  <link name="base">
    <inertial>
      <mass value="50.455"/>
      <origin xyz="0 0 0.127"/>
      <inertia ixx="1.391" ixy="0.004" ixz="0.0" iyy="6.853" iyz="0.0" izz="6.125"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.127"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/bases/xls/summit_xls_chassis.stl"/>
      </geometry>
      <material name="black">
        <color rgba="0.0 0.0 0.0 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0 " xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/bases/xls/summit_xls_chassis.dae"/>
      </geometry>
    </collision>
  </link>


  <!-- CHAPAS INOX TAPAS -->
  <link name="summit_xl_base_chapa">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0.127"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/bases/xls/summit_xls_chapas_inox_tapas.stl"/>
      </geometry>
      <material name="darkgrey">
        <color rgba="0.5 0.5 0.5 1"/>
      </material>
    </visual>
  </link>
  <joint name="summit_xl_base_tapa_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.0 0 0.38062"/>
    <parent link="base"/>
    <child link="summit_xl_base_chapa"/>
  </joint>
  <!-- LOGO ROBOTNIK IZQUIERDO -->
  <link name="summit_xl_base_logo_left">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/bases/xls/robotnik_logo_chasis.stl"/>
      </geometry>
      <material name="darkgrey">
        <color rgba="0.5 0.5 0.5 1"/>
      </material>
    </visual>
  </link>
  <joint name="summit_xl_base_logo_left_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.0 0.28931 0.41762"/>
    <parent link="base"/>
    <child link="summit_xl_base_logo_left"/>
  </joint>
  <!-- LOGO ROBOTNIK DERECHO -->
  <link name="summit_xl_base_logo_right">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/bases/xls/robotnik_logo_chasis.stl"/>
      </geometry>
      <material name="darkgrey">
        <color rgba="0.5 0.5 0.5 1"/>
      </material>
    </visual>
  </link>
  <joint name="summit_xl_base_logo_right_joint" type="fixed">
    <origin rpy="0 0 3.141592653589793" xyz="0.0 -0.28931 0.41762"/>
    <parent link="base"/>
    <child link="summit_xl_base_logo_right"/>
  </joint>
  <!-- LOGO ROBOTNIK FRONTAL -->
  <link name="summit_xl_base_logo_front">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/bases/xls/robotnik_logo_chasis.stl"/>
      </geometry>
      <material name="darkgrey">
        <color rgba="0.5 0.5 0.5 1"/>
      </material>
    </visual>
  </link>
  <joint name="summit_xl_base_logo_front_joint" type="fixed">
    <origin rpy="0 0 -1.5707963267948966" xyz="0.345 0.0 0.127"/>
    <parent link="base"/>
    <child link="summit_xl_base_logo_front"/>
  </joint>
  <!-- LOGO ROBOTNIK TRASERO -->
  <link name="summit_xl_base_logo_rear">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/bases/xls/robotnik_logo_chasis.stl"/>
      </geometry>
      <material name="darkgrey">
        <color rgba="0.5 0.5 0.5 1"/>
      </material>
    </visual>
  </link>
  <joint name="summit_xl_base_logo_rear_joint" type="fixed">
    <origin rpy="0 0 1.5707963267948966" xyz="-0.345 0.0 0.127"/>
    <parent link="base"/>
    <child link="summit_xl_base_logo_rear"/>
  </joint>
  <!-- DOCKING CONTACT -->
  <link name="summit_xl_base_docking_contact">
    </link>
  <joint name="summit_xl_base_docking_joint" type="fixed">
    <origin rpy="0 0 0" xyz="0.35 0 0.217"/>
    <parent link="base"/>
    <child link="summit_xl_base_docking_contact"/>
  </joint>
  <joint name="summit_xl_front_right_wheel_joint" type="fixed">
    <parent link="base"/>
    <child link="summit_xl_front_right_wheel_link"/>
    <origin rpy="0 0 0" xyz="0.2225 -0.2045 0.127"/>
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <limit effort="100" velocity="100"/>
    <joint_properties damping="0.0" friction="0.0"/>
  </joint>
  <link name="summit_xl_front_right_wheel_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/wheels/omni_wheel_2.dae"/>
      </geometry>
      <material name="darkgrey">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <!--sphere radius="${omni_wheel_radius}" /-->
        
          <!-- <xacro:if value="${reflect}">
              <mesh filename="package://mujoco_ros2/models/meshes/wheels/omni_wheel_1.dae" />
          </xacro:if>
          <xacro:unless value="${reflect}">
              <mesh filename="package://mujoco_ros2/models/meshes/wheels/omni_wheel_2.dae" />
					</xacro:unless> -->

        <cylinder length="0.087" radius="0.127"/>
        <!-- sphere radius="${omni_wheel_radius}"/-->
      </geometry>
    </collision>
    <inertial>
      <mass value="6.5"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0303095" ixy="0" ixz="0" iyy="0.05241925" iyz="0" izz="0.0303095"/>
    </inertial>
  </link>
  <!-- Transmission is important to link the joints and the controller (see summit_xl_control/config/summit_xl_control.yaml)-->
  <transmission name="summit_xl_front_right_wheel_joint_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="summit_xl_front_right_wheel_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="summit_xl_front_right_wheel_joint_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="summit_xl_front_right_wheel_link">
    <mu1 value="0.0"/>
    <mu2 value="0.0"/>
    <kp value="10000000.0"/>
    <kd value="1.0"/>
    <fdir1 value="1 0 0"/>
    <turnGravityOff>false</turnGravityOff>
  </gazebo>
  <joint name="summit_xl_front_left_wheel_joint" type="fixed">
    <parent link="base"/>
    <child link="summit_xl_front_left_wheel_link"/>
    <origin rpy="0 0 0" xyz="0.2225 0.2045 0.127"/>
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <limit effort="100" velocity="100"/>
    <joint_properties damping="0.0" friction="0.0"/>
  </joint>
  <link name="summit_xl_front_left_wheel_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/wheels/omni_wheel_1.dae"/>
      </geometry>
      <material name="darkgrey">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <!--sphere radius="${omni_wheel_radius}" /-->
        
          <!-- <xacro:if value="${reflect}">
              <mesh filename="package://mujoco_ros2/models/meshes/wheels/omni_wheel_1.dae" />
          </xacro:if>
          <xacro:unless value="${reflect}">
              <mesh filename="package://mujoco_ros2/models/meshes/wheels/omni_wheel_2.dae" />
					</xacro:unless> -->

        <cylinder length="0.087" radius="0.127"/>
        <!-- sphere radius="${omni_wheel_radius}"/-->
      </geometry>
    </collision>
    <inertial>
      <mass value="6.5"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0303095" ixy="0" ixz="0" iyy="0.05241925" iyz="0" izz="0.0303095"/>
    </inertial>
  </link>
  <!-- Transmission is important to link the joints and the controller (see summit_xl_control/config/summit_xl_control.yaml)-->
  <transmission name="summit_xl_front_left_wheel_joint_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="summit_xl_front_left_wheel_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="summit_xl_front_left_wheel_joint_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="summit_xl_front_left_wheel_link">
    <mu1 value="0.0"/>
    <mu2 value="0.0"/>
    <kp value="10000000.0"/>
    <kd value="1.0"/>
    <fdir1 value="1 0 0"/>
    <turnGravityOff>false</turnGravityOff>
  </gazebo>
  <joint name="summit_xl_back_left_wheel_joint" type="fixed">
    <parent link="base"/>
    <child link="summit_xl_back_left_wheel_link"/>
    <origin rpy="0 0 0" xyz="-0.2225 0.2045 0.127"/>
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <limit effort="100" velocity="100"/>
    <joint_properties damping="0.0" friction="0.0"/>
  </joint>
  <link name="summit_xl_back_left_wheel_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/wheels/omni_wheel_1.dae"/>
      </geometry>
      <material name="darkgrey">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <!--sphere radius="${omni_wheel_radius}" /-->
        
          <!-- <xacro:if value="${reflect}">
              <mesh filename="package://mujoco_ros2/models/meshes/wheels/omni_wheel_1.dae" />
          </xacro:if>
          <xacro:unless value="${reflect}">
              <mesh filename="package://mujoco_ros2/models/meshes/wheels/omni_wheel_2.dae" />
					</xacro:unless> -->

        <cylinder length="0.087" radius="0.127"/>
        <!-- sphere radius="${omni_wheel_radius}"/-->
      </geometry>
    </collision>
    <inertial>
      <mass value="6.5"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0303095" ixy="0" ixz="0" iyy="0.05241925" iyz="0" izz="0.0303095"/>
    </inertial>
  </link>
  <!-- Transmission is important to link the joints and the controller (see summit_xl_control/config/summit_xl_control.yaml)-->
  <transmission name="summit_xl_back_left_wheel_joint_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="summit_xl_back_left_wheel_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="summit_xl_back_left_wheel_joint_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="summit_xl_back_left_wheel_link">
    <mu1 value="0.0"/>
    <mu2 value="0.0"/>
    <kp value="10000000.0"/>
    <kd value="1.0"/>
    <fdir1 value="1 0 0"/>
    <turnGravityOff>false</turnGravityOff>
  </gazebo>
  <joint name="summit_xl_back_right_wheel_joint" type="fixed">
    <parent link="base"/>
    <child link="summit_xl_back_right_wheel_link"/>
    <origin rpy="0 0 0" xyz="-0.2225 -0.2045 0.127"/>
    <axis rpy="0 0 0" xyz="0 1 0"/>
    <limit effort="100" velocity="100"/>
    <joint_properties damping="0.0" friction="0.0"/>
  </joint>
  <link name="summit_xl_back_right_wheel_link">
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/wheels/omni_wheel_2.dae"/>
      </geometry>
      <material name="darkgrey">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="1.5707963267948966 0 0" xyz="0 0 0"/>
      <geometry>
        <!--sphere radius="${omni_wheel_radius}" /-->
        
          <!-- <xacro:if value="${reflect}">
              <mesh filename="package://summit_xl_description/meshes/wheels/omni_wheel_1.dae" />
          </xacro:if>
          <xacro:unless value="${reflect}">
              <mesh filename="package://summit_xl_description/meshes/wheels/omni_wheel_2.dae" />
					</xacro:unless> -->

        <cylinder length="0.087" radius="0.127"/>
        <!-- sphere radius="${omni_wheel_radius}"/-->
      </geometry>
    </collision>
    <inertial>
      <mass value="6.5"/>
      <origin xyz="0 0 0"/>
      <inertia ixx="0.0303095" ixy="0" ixz="0" iyy="0.05241925" iyz="0" izz="0.0303095"/>
    </inertial>
  </link>
  <!-- Transmission is important to link the joints and the controller (see summit_xl_control/config/summit_xl_control.yaml)-->
  <transmission name="summit_xl_back_right_wheel_joint_trans">
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="summit_xl_back_right_wheel_joint">
      <hardwareInterface>hardware_interface/VelocityJointInterface</hardwareInterface>
    </joint>
    <actuator name="summit_xl_back_right_wheel_joint_motor">
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <gazebo reference="summit_xl_back_right_wheel_link">
    <mu1 value="0.0"/>
    <mu2 value="0.0"/>
    <kp value="10000000.0"/>
    <kd value="1.0"/>
    <fdir1 value="1 0 0"/>
    <turnGravityOff>false</turnGravityOff>
  </gazebo>
  <joint name="summit_xl_imu_joint" type="fixed">
    <axis xyz="1 0 0"/>
    <!--origin xyz="0 0 0.2825"/-->
    <origin rpy="0 0 0" xyz="0.127 -0.129 0.212"/>
    <parent link="base"/>
    <child link="summit_xl_imu_link"/>
  </joint>
  <link name="summit_xl_imu_link">
    <inertial>
      <origin rpy="0 0 0" xyz="0.0 0 0.01"/>
      <mass value="0.005"/>
      <inertia ixx="8.333333333333334e-10" ixy="0" ixz="0" iyy="8.333333333333334e-10" iyz="0" izz="8.333333333333334e-10"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <box size="0.001 0.001 0.001"/>
      </geometry>
    </visual>
    <!--collision>
	        <origin rpy="0 0 0" xyz="0 0 0"/>
	        <geometry>
	          <box size="0.001 0.001 0.001"/>
	        </geometry>
	      </collision-->
  </link>

  <joint name="panda_to_world" type="fixed">
    <parent link="base"/>
    <child link="panda_link0"/>
    <origin rpy="0 0 0" xyz="0.2 0 0.5"/> <!--이거 z가 0.5로 받앗엇음-->
  </joint>
  <link name="panda_link0">
  	<inertial>
      <mass value="2.3966"/>
      <origin xyz="-0.041018 -0.00014 0.049974" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link0.dae"/>
      </geometry>
      <inertia ixx="0.0090" ixy="0.0000" ixz="0.0020" iyy="0.0115" iyz="0.0000" izz="0.0085"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link0.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <link name="panda_link1">
  	<inertial>
      <mass value="2.9274653454"/>
      <origin xyz="0.003875 0.002081 -0.04762" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link1.dae"/>
      </geometry>
      <inertia ixx="0.023927316485107913" ixy="1.3317903455714081e-05" ixz="-0.00011404774918616684" iyy="0.0224821613275756" iyz="-0.0019950320628240115" izz="0.006350098258530016"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link1.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="joint1" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.7437" soft_upper_limit="2.7437"/>
    <origin rpy="0 0 0" xyz="0 0 0.333"/>
    <parent link="panda_link0"/>
    <child link="panda_link1"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-2.7437" upper="2.7437" velocity="2.1750"/>
  </joint>
  <link name="panda_link2">
  	<inertial>
      <mass value="2.9355370338"/>
      <origin xyz="-0.003141 -0.02872 0.003495" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link2.dae"/>
      </geometry>
      <inertia ixx="0.041938946257609425" ixy="0.00020257331521090626" ixz="0.004077784227179924" iyy="0.02514514885014724" iyz="-0.0042252158006570156" izz="0.06170214472888839"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link2.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="joint2" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-1.7837" soft_upper_limit="1.7837"/>
    <origin rpy="-1.57079632679 0 0" xyz="0 0 0"/>
    <parent link="panda_link1"/>
    <child link="panda_link2"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-1.7837" upper="1.7837" velocity="2.1750"/>
  </joint>
  <link name="panda_link3">
	  <inertial>
      <mass value="2.2449013699"/>
      <origin xyz="2.7518e-2 3.9252e-2 -6.6502e-2" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link3.dae"/>
      </geometry>
      <inertia ixx="0.02410142547240885" ixy="0.002404694559042109" ixz="-0.002856269270114313" iyy="0.01974053266708178" iyz="-0.002104212683891874" izz="0.019044494482244823"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link3.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="joint3" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.9007" soft_upper_limit="2.9007"/>
    <origin rpy="1.57079632679 0 0" xyz="0 -0.316 0"/>
    <parent link="panda_link2"/>
    <child link="panda_link3"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-2.9007" upper="2.9007" velocity="2.1750"/>
  </joint>
  <link name="panda_link4">
  	<inertial>
      <mass value="2.6155955791"/>
      <origin xyz="-5.317e-2 1.04419e-1 2.7454e-2" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link4.dae"/>
      </geometry>
      <inertia ixx="0.03452998321913202" ixy="0.01322552265982813" ixz="0.01015142998484113" iyy="0.028881621933049058" iyz=" -0.0009762833870704552" izz="0.04125471171146641"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link4.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="joint4" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-3.0421" soft_upper_limit="-0.1518"/>
    <origin rpy="1.57079632679 0 0" xyz="0.0825 0 0"/>
    <parent link="panda_link3"/>
    <child link="panda_link4"/>
    <axis xyz="0 0 1"/>
    <limit effort="87" lower="-3.0421" upper="-0.1518" velocity="2.1750"/>
  </joint>
  <link name="panda_link5">
  	<inertial>
      <mass value="2.3271207594"/>
      <origin xyz="-1.1953e-2 4.1065e-2 -3.8437e-2" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link5.dae"/>
      </geometry>
      <inertia ixx="0.051610278463662895" ixy="-0.005715173387783472" ixz="-0.0035673167625872135" iyy="0.04787729713371481" iyz="0.010673985108535986" izz="0.016423625579357254"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link5.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="joint5" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-2.8065" soft_upper_limit="2.8065"/>
    <!-- <safety_controller k_position="100.0" k_velocity="40.0"/> -->
    <origin rpy="-1.57079632679 0 0" xyz="-0.0825 0.384 0"/>
    <parent link="panda_link4"/>
    <child link="panda_link5"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-2.8065" upper="2.8065" velocity="2.6100"/>
    <!-- <limit effort="12" velocity="2.6100"/> -->
  </joint>
  <link name="panda_link6">
  	<inertial>
      <mass value="1.8170376524"/>
      <origin xyz="6.0149e-2 -1.4117e-2 -1.0517e-2" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link6.dae"/>
      </geometry>
      <inertia ixx="0.005412333594383447" ixy="0.006193456360285834" ixz="0.0014219289306117652" iyy="0.014058329545509979" iyz="-0.0013140753741120031" izz="0.016080817924212554"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link6.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="joint6" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="0.5445" soft_upper_limit="4.5169"/>
    <origin rpy="1.57079632679 0 0" xyz="0 0 0"/>
    <parent link="panda_link5"/>
    <child link="panda_link6"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="0.5445" upper="4.5169" velocity="2.6100"/>
  </joint>
  <link name="panda_link7">
  	<inertial>
      <mass value="0.6271432862"/>
      <origin xyz="1.0517e-2 -4.252e-3 6.1597e-2" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link7.dae"/>
      </geometry>
      <inertia ixx="0.00021092389150104718" ixy="-2.433299114461931e-05" ixz="4.564480393778983e-05" iyy="0.00017718568002411474" iyz="8.744070223226438e-05" izz="5.993190599659971e-05"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/link7.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="joint7" type="revolute">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-3.0159" soft_upper_limit="3.0159"/>
    <origin rpy="1.57079632679 0 0" xyz="0.088 0 0"/>
    <parent link="panda_link6"/>
    <child link="panda_link7"/>
    <axis xyz="0 0 1"/>
    <limit effort="12" lower="-3.0159" upper="3.0159" velocity="2.6100"/>
  </joint>


  <link name="panda_link8"/>


  <joint name="panda_joint8" type="fixed">
    <safety_controller k_position="100.0" k_velocity="40.0" soft_lower_limit="-0.0175" soft_upper_limit="3.0421"/>
    <origin rpy="0 0 0" xyz="0 0 0.107"/>
    <parent link="panda_link7"/>
    <child link="panda_link8"/>
    <axis xyz="0 0 0"/>
    <!-- <limit effort="12" lower="-0.0873" upper="3.8223" velocity="2.6100"/> -->
  </joint>

  <joint name="panda_hand_joint" type="fixed">
    <parent link="panda_link8"/>
    <child link="panda_hand"/>
    <origin rpy="0 0 -0.785398163397" xyz="0 0 0"/>
  </joint>
  <link name="panda_hand">
    <inertial>
      <mass value="0.73"/>
      <origin xyz="-0.01 0 0.03" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/hand.dae"/>
      </geometry>
      <inertia ixx="0.001" ixy="0" ixz="0" iyy="0.0025" iyz="0" izz="0.0017"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/hand.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <link name="panda_leftfinger">
    <inertial>
      <mass value="0.015"/>
      <origin xyz="0 0 0" rpy="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/finger.dae"/>
      </geometry>
      <inertia ixx="2.375e-6" ixy="0" ixz="0" iyy="2.375e-6" iyz="0" izz="7.5e-7"/>
    </inertial>
    <visual>
      <origin rpy="0 0 0" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/finger.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <link name="panda_rightfinger">
    <inertial>
      <mass value="0.015"/>
      <origin xyz="0 0 0" rpy="0 0 3.14"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/finger.dae"/>
      </geometry>
      <inertia ixx="2.375e-6" ixy="0" ixz="0" iyy="2.375e-6" iyz="0" izz="7.5e-7"/>
    </inertial>
    <visual>
      <origin rpy="0 0 3.14" xyz="0 0 0"/>
      <geometry>
        <mesh filename="package://mujoco_ros2/models/meshes/visual/finger.dae"/>
      </geometry>
      <material name="">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <joint name="panda_finger_joint1" type="fixed">
  <!-- <joint name="panda_finger_joint1" type="prismatic"> -->
    <parent link="panda_hand"/>
    <child link="panda_leftfinger"/>
    <origin rpy="0 0 0" xyz="0 0 0.0584"/>
    <axis xyz="0 1 0"/>
    <limit effort="20" lower="0.0" upper="0.04" velocity="0.2"/>
  </joint>
  <joint name="panda_finger_joint2" type="fixed">
  <!-- <joint name="panda_finger_joint2" type="prismatic"> -->
    <parent link="panda_hand"/>
    <child link="panda_rightfinger"/>
    <origin rpy="0 0 0" xyz="0 0 0.0584"/>
    <axis xyz="0 -1 0"/>
    <limit effort="20" lower="0.0" upper="0.04" velocity="0.2"/>
    <mimic joint="panda_finger_joint1"/>
  </joint>

</robot>


